std/algorithm/searching.d

```diff
@@ -1768,14 +1768,14 @@ if (isForwardRange!R1 && isForwardRange!R2
 
 /// ditto
 R1 find(alias pred = "a == b", R1, R2)(R1 haystack, R2 needle)
-if (isRandomAccessRange!R1 && isBidirectionalRange!R2
+if (isRandomAccessRange!R1 && hasLength!R1 && hasSlicing!R1 && isBidirectionalRange!R2
         && is(typeof(binaryFun!pred(haystack.front, needle.front)) : bool))
```

Comment:
This is a breaking change, but there are probably very few random access ranges without length and slicing (if at all).